File: app/src/main/java/com/ternaryop/photoshelf/adapter/SelectionArrayViewHolder.java
package com.ternaryop.photoshelf.adapter;

import android.support.v7.widget.RecyclerView;

/**
 * Created by dave on 13/04/16.
 *
 * Notify changes to the adapter delegate
 */
public class SelectionArrayViewHolder<T extends RecyclerView.ViewHolder> extends SelectionArray {
    private RecyclerView.Adapter<T> adapter;

    public SelectionArrayViewHolder(RecyclerView.Adapter<T> adapter) {
        this.adapter = adapter;
    }

    @Override
    public void toggle(int position) {
        super.toggle(position);
        adapter.notifyItemChanged(position);
    }

    @Override
    public void setSelected(int position, boolean selected) {
        super.setSelected(position, selected);
        adapter.notifyItemChanged(position);
    }

    @Override
    public void clear() {
        super.clear();
        adapter.notifyDataSetChanged();
    }
}

File: core/src/main/java/com/ternaryop/photoshelf/adapter/SelectionArrayViewHolder.kt
package com.ternaryop.photoshelf.adapter

import androidx.recyclerview.widget.RecyclerView

/**
 * Created by dave on 13/04/16.
 *
 * Notify changes to the adapter delegate
 */
class SelectionArrayViewHolder<T : RecyclerView.ViewHolder>(
    private val adapter: RecyclerView.Adapter<T>
) : SelectionArray() {

    override fun toggle(position: Int) {
        super.toggle(position)
        adapter.notifyItemChanged(position)
    }

    override fun setSelected(position: Int, selected: Boolean) {
        super.setSelected(position, selected)
        adapter.notifyItemChanged(position)
    }

    override fun clear() {
        super.clear()
        adapter.notifyItemRangeChanged(0, adapter.itemCount)
    }

    override fun setSelectedRange(start: Int, end: Int, selected: Boolean) {
        super.setSelectedRange(start, end, selected)
        adapter.notifyDataSetChanged()
    }
}
